module/Finna/src/Finna/RecordDriver/SolrLrmi.php

```diff
@@ -33,9 +37,465 @@
+     *
+     * @return array|boolean
+     */
+    public function getExternalRatingLink()
+    {
+        // Only AOE records supported currently.
+        if ('aoe' !== $this->getDataSource()
+            || !$link = $this->getExternalLink()
+        ) {
+            return false;
+        }
+        return [
+            'key' => 'add_aoe_rating_html',
+            'link' => $link
+        ];
+    }
+
+    /**
+     * Get educational subjects
+     *
+     * @return array
+     */
+    public function getEducationalSubjects()
+    {
+        return $this->fields['educational_subject_str_mv'] ?? [];
+    }
+
+    /**
+     * Get educational material type
+     *
+     * @return array
+     */
+    public function getEducationalMaterialType()
+    {
+        return $this->fields['educational_material_type_str_mv'] ?? [];
+    }
+
+    /**
+     * Get topics
+     *
+     * @param string $type defaults to /onto/yso/
+     *
+     * @return array
+     */
+    public function getTopics($type = '/onto/yso/')
+    {
+        $xml = $this->getXmlRecord();
+        $topics = [];
+        foreach ($xml->about as $about) {
+            $thing = $about->thing;
+            $name = trim((string)$thing->name);
+            if ($name && strpos((string)$thing->identifier, $type) !== false
+            ) {
+                $topics[] = $name;
+            }
+        }
+        return $topics;
+    }
+
+    /**
+     * Is the provided filetype allowed for download?
+     *
+     * @param string $format file format
+     *
+     * @return boolean
+     */
+    protected function isDownloadableFileFormat($format)
+    {
+        return in_array($format, $this->downloadableFileFormats);
+    }
+
+    /**
+     * Get file format
+     *
+     * @param string $filename file name
+     *
+     * @return string
+     */
+    protected function getFileFormat($filename)
+    {
+        return pathinfo($filename)['extension'] ?? '';
+    }
+
+    /**
+     * Return an array of image URLs associated with this record with keys:
+     * - url         Image URL
+     * - description Description text
+     * - rights      Rights
+     *   - copyright   Copyright (e.g. 'CC BY 4.0') (optional)
+     *   - description Human readable description (array)
+     *   - link        Link to copyright info
+     *
+     * @param string $language   Language for copyright information
+     * @param bool   $includePdf Whether to include first PDF file when no image
+     * links are found
+     *
+     * @return mixed
+     */
+    public function getAllImages($language = 'fi', $includePdf = true)
+    {
+        $xml = $this->getXmlRecord();
+        $result = [];
+        $images = ['image/png', 'image/jpeg'];
+        foreach ($xml->description as $desc) {
+            $attr = $desc->attributes();
+            $format = trim((string)($attr['format'] ?? ''));
+            if ($format && in_array($format, $images)) {
```

Comment:
Tähän lisäksi tarkistukset `!$this->urlBlocked('url')` ja `$this->isUrlLoadable('url', $this->getUniqueID())`.